src/main/java/mate/academy/dao/MySqlBookDao.java

```diff
@@ -0,0 +1,123 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.lib.ConnectionUtil;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.model.DataProcessingException;
+
+@Dao
+public class MySqlBookDao implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO book (title, price) VALUES (?,?)";
+        int idIndex = 1;
+        int titleIndex = 1;
+        int priceIndex = 2;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(titleIndex, book.getTitle());
+            statement.setBigDecimal(priceIndex, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to insert at least 1 row, "
+                        + "but inserted 0 rows, with id = " + book.getId());
+            }
```

Comment:
you can insert 0 rows. it's not a mistake. You don't need to check affected rows.